mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/mail/InMemoryAnnotationMapper.java

```diff
@@ -128,7 +128,7 @@ public void insertAnnotation(MailboxId mailboxId, MailboxAnnotation mailboxAnnot
     public void deleteAnnotation(MailboxId mailboxId, MailboxAnnotationKey key) {
         lock.writeLock().lock();
         try {
-            mailboxesAnnotations.remove(mailboxId, key.asString());
+            mailboxesAnnotations.remove((InMemoryId)mailboxId, key.asString());
```

Comment:
Why this cast ?

Reply:
We know the exact type so we can cast safely (it's done in others methods).
It prevents from calling `remove(Object, Object)` method when the target datastructure is actually typed.

Reply:
space